app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -230,6 +236,11 @@ class HomeFragment : Fragment() {
             ::dispatchModeChanges
         )
 
+        val blocklistProvider = LocalHomescreenBlocklistProvider(
+            lifecycleScope,
+            requireContext().homescreenBlocklistDataStore
+        )
+        val initialBlocklist = runBlockingIncrement { blocklistProvider.blocklist.first() }
```

Comment:
I think this might've been more complicated because of the introduction of the async preference datastore. I am wondering if this would just be simpler if we added a string preference to `Settings`.

Reply:
I think you're right, that's easiest for now. I'd be interested in working to develop better patterns around the use of `DataStore`s eventually, but this is seeming like a awkward way to slot this in here.

Reply:
Ya, I looked into this briefly in https://github.com/mozilla-mobile/fenix/issues/19197 and https://github.com/mozilla-mobile/fenix/pull/19999. My discussion with @jonalmeida concluded with the need to develop an incremental and long term plan to migrate existing shared preferences (for example, fragment by fragment to ensure app stability), agree on a plan, pattern and troubleshoot various scenarios via an RFC.

Reply:
That said there's no problem with using datastore for new features as you see for Pocket, but perhaps maybe something for preferences you can imagine you might just want 1 datastore to hold all preferences.

Reply:
👍 I'd definitely be interested in being included in future discussions about it